flink-core/src/main/java/org/apache/flink/api/connector/source/SourceReaderContext.java

```diff
@@ -39,7 +41,16 @@ public interface SourceReaderContext {
      */
     String getLocalHostName();
 
-    /** @return The index of this subtask. */
+    /**
+     * Get the index of this subtask.
+     *
+     * @deprecated This method is deprecated since Flink 1.19. All metadata about the task should be
+     *     provided uniformly by {@link #getTaskInfo()}.
+     * @see <a
+     *     href="https://cwiki.apache.org/confluence/display/FLINK/FLIP-382%3A+Unify+the+Provision+of+Diverse+Metadata+for+Context-like+APIs">
+     *     FLIP-382: Unify the Provision of Diverse Metadata for Context-like APIs </a>
+     */
+    @Deprecated
     int getIndexOfSubtask();
```

Comment:
Same here for the deprecated methods.

Reply:
For these deprecated methods in the modified interfaces, I've changed them from abstract to default and provide default implementations.